### PR TITLE
[UNI-179] Generic port for rosserial_server -> serial_node

### PR DESCRIPTION
**Pull request by javijurado**
### Changes:

- Set C++17 as a default compiler to be able to use filesystem libraries.
- `Rosserial_server serial_node` can look for port is a partial name is given. In the partial port name, it has to contain '*'.  **Will return the first match found**

eg 1:

_input:_ /dev/ttyACM*
_output:_  /dev/ttyACM0

eg 2:
`rosrun rosserial_server serial_node _port:=/dev/serial/by-id/usb-Espressif* _baud:=115200`

image

### CODE REVIEW

**Review by meyerj — COMMENTED** [review]
String parsing is always error-prone, and there is a simpler and more generic approach: [glob](https://man7.org/linux/man-pages/man3/glob.3.html). Unforutunately the C++ STL does not support it directly and an [extra library dependencies](https://github.com/p-ranav/glob/tree/master) seems overkill to me, but the C API is simple enough. More details on Slack.

The `*` may not necessarily appear at the end of the `port` string.

**Comment by javijurado** [comment]
After the changes:
image

File: rosserial_server/include/rosserial_server/utils.h [hunk 1]
```diff
@@ -0,0 +1,35 @@
+/******************************************************************************
+ *  Copyright (c) 2024, Intermodalics BVBA                                    *
+ *  All rights reserved.                                                      *
+ ******************************************************************************/
+
+#ifndef ROSSERIAL_SERVER_UTILS_H
+#define ROSSERIAL_SERVER_UTILS_H
+
+#include <filesystem>
```
Comment (meyerj) [inline rosserial_server/include/rosserial_server/utils.h]:
```suggestion
```

Not required anymore, and neither is setting `CMAKE_CXX_STANDARD` explicitly in `CMakeLists.txt`.

File: rosserial_server/include/rosserial_server/utils.h [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+#include <glob.h>
+
+inline std::string getPortNameByPartialPortName(const std::string& port_name){
```
Comment (meyerj) [inline rosserial_server/include/rosserial_server/utils.h]: nit: rename `getPortNameByPartialPortName()` -> `getPortNameFromPattern()`?

A wildcard "pattern" (the terminology from [glob](https://linux.die.net/man/7/glob)) like `/dev/serial/by-id/usb-Espressif*` is a bit more generic and not just a "partial port name".

**Review by meyerj — APPROVED** [review]
Apart from the nit comments below, LGTM!

File: rosserial_server/include/rosserial_server/utils.h (hunk 2 above)
Comment (javijurado) [reply rosserial_server/include/rosserial_server/utils.h]: Make more sense to use that name. Thanks for the suggestion

**Pull request merged**